app/boards/shields/waterfowl/waterfowl.dtsi

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <dt-bindings/zmk/matrix_transform.h>
+
+/ {
+	chosen {
+		zmk,kscan = &kscan0;
+		zmk,matrix_transform = &default_transform;
+	};
+
+	default_transform: keymap_transform_0 {
+		compatible = "zmk,matrix-transform";
+		columns = <16>;
+		rows = <4>;
+// | MX5  | MX4  | MX3  | MX2  | MX1  |                               | MX1  | MX2  | MX3  | MX4  | MX5  |
+// | MX10 | MX9  | MX8  | MX7  | MX6  |                               | MX6  | MX7  | MX8  | MX9  | MX10 |
+// | MX15 | MX14 | MX13 | MX12 | MX11 |								  | MX11 | MX12 | MX13 | MX14 | MX15 |
+//               | MX20 | MX19 | MX18 | MX17 | MX16 |   | MX16 | MX17 | MX18 | MX19 | MX20 |
+		map = <
+RC(0,0) RC(0,1) RC(0,2) RC(0,3) RC(0,4)                                 RC(0,5) RC(0,6) RC(0,7) RC(0,8) RC(0,9)
+RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4)                                 RC(1,5) RC(1,6) RC(1,7) RC(1,8) RC(1,9)
+RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4)									RC(2,5) RC(2,6) RC(2,7) RC(2,8) RC(2,9)
+RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,4)   								RC(3,5) RC(3,6) RC(3,7) RC(3,8) RC(3,9)
+		>;
+	};
+
+	kscan0: kscan {
+		compatible = "zmk,kscan-gpio-matrix";
+		label = "KSCAN";
+
+		diode-direction = "col2row";
+		row-gpios
+			= <&pro_micro 4 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&pro_micro 5 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&pro_micro 6 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&pro_micro 7 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			;
+		
+	};
+
+	left_encoder: encoder_left {    //roller
+		compatible = "alps,ec11";
+		label = "LEFT_ENCODER";
+		a-gpios = <&pro_micro 8 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+		b-gpios = <&pro_micro 9 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+		resolution = <4>;
+	};
+
+	right_encoder: encoder_right {  //Standard encoder on left half
+		compatible = "alps,ec11";
+		label = "RIGHT_ENCODER";
+		a-gpios = <&pro_micro 14 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+		b-gpios = <&pro_micro 16 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+		resolution = <2>;
+	};
+
+	sensors {
+		compatible = "zmk,keymap-sensors";
+		sensors = <&left_encoder &right_encoder>;
+	};
+
+	// TODO: RGB node(s)
+};
+
+&pro_micro_i2c {
+	status = "okay";
+
+	ssd1306@3c {
```

Comment:
You need to add a label to this node (`oled: ssd1306@3c`) then use it in the `chosen` node above like `zephyr,display= &oled;`